Add support for THERMOplate.getCOLD and DAQC2plate.getFREQ

charkins opened the pull request.

charkins committed:
Support getting cold junction temperature from THERMOplate.

charkins committed:
Support for frequency counter on DAQC2plate.

charkins committed:
Set LED color on DAQC(2) plate.

charkins committed:
Split handling of setLED for DAQC and DAQC2.

charkins committed:
THERMOplate LED control.

charkins committed:
Change arg from "value" to "color".

mharsch commented:
@charkins thanks for working on this.  Sorry for the delay -- I'll get these changes merged  tonight.

charkins commented:
No worries. I forgot I had these pull requests open and it looks like I made the changes on master branch, so there are some new commits here and in the node-red-contrib-pi-plates dealing with LEDs. If you'd prefer, I could break these up into branches with separate pull requests. 

For THERMOplate, the API matches the existing RELAYplate API, with the exception that it returns the actual LED state on toggle rather than "UNKNOWN". For DAQC2plate, the API matches the pi-plates python API, accepting a string color for the LED. For DAQCplate, the API I added accepts string colors (off, red, green, yellow), where "yellow" turns both the red and green on. It might be better to expose a node api that more closely matches the pi-plates API.

On the node-red side, I added a single ppLED node. For THERMOplate and RELAYplate, msg.payload=='off' on the input turns the LED off, otherwise it turns it on. For DAQC(2)plate, it passes the msg.payload as the color to the API here.
